src/Servers/HttpSys/test/FunctionalTests/Listener/ResponseHeaderTests.cs

```diff
@@ -221,7 +220,28 @@ public async Task ResponseHeaders_HTTP10KeepAliveRequest_Gets11Close()
             HttpResponseMessage response = await responseTask;
             response.EnsureSuccessStatusCode();
             Assert.Equal(new Version(1, 1), response.Version);
-            Assert.True(response.Headers.ConnectionClose.Value);
+            Assert.Null(response.Headers.ConnectionClose);
+        }
+    }
+
+    [ConditionalFact]
+    public async Task ResponseHeaders_HTTP10KeepAliveRequest_ChunkedTransferEncoding_Gets11Close()
+    {
+        string address;
+        using (var server = Utilities.CreateHttpServer(out address))
+        {
+            Task<HttpResponseMessage> responseTask = SendRequestAsync(address, usehttp11: false, sendKeepAlive: true);
+
+            var context = await server.AcceptAsync(Utilities.DefaultTimeout).Before(responseTask);
+            context.Response.Headers["Transfer-Encoding"] = new string[] { "chunked" };
+            var responseBytes = Encoding.ASCII.GetBytes("10\r\nManually Chunked\r\n0\r\n\r\n");
+            await context.Response.Body.WriteAsync(responseBytes, 0, responseBytes.Length);
+            context.Dispose();
+
+            HttpResponseMessage response = await responseTask;
+            response.EnsureSuccessStatusCode();
+            Assert.Equal(new Version(1, 1), response.Version);
+            Assert.True(response.Headers.TransferEncodingChunked.HasValue, "Chunked");
```

Comment:
This should be checking for close?

Reply:
Isn't that on the next line? `Assert.True(response.Headers.ConnectionClose.Value);`?

Reply:
Fixed now.